src/main/java/com/bnorm/auto/dispatch/internal/Writer.java

```diff
@@ -108,9 +112,38 @@ public static JavaFile write(MethodClassDescriptor methodClassDescriptor) {
         codeBlockBuilder.addStatement("throw new NullPointerException($S)", "dispatch value == null");
         first = false;
       }
-      for (MultiDescriptor descriptor : methodDescriptor.multiMethods()) {
+
+      // Work with multiMethod dispatch values as a list
+      // This allows a multiMethod to accept several dispatch values
+      Function<MultiDescriptor, List<?>> toDispatchValues = descriptor ->
+          descriptor.value().getValue() instanceof List<?> ? (List<?>) descriptor.value().getValue()
```

Comment:
That's what I thought initially, but it turns out it's a List. I'll move this to the discover phase